src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition2Test.java

```diff
@@ -897,7 +904,6 @@ public void cpsScriptNonAdminConfiguration() throws Exception {
 
         jenkins.submit(config);
 
-        assertEquals(1, ScriptApproval.get().getPendingScripts().size());
```

Comment:
Removed because this sometimes causes test failures locally, it was added in https://github.com/jenkinsci/workflow-cps-plugin/commit/6bd4e8bf837ba1df4394efdd003c70bb1b9ac33f by me. This assertion doesn't make much sense on its own, checking that specific script didn't get approved on the next line should be enough.

Reply:
Does it fail because the value is 0 or because it is more than 1? And does it fail intermittently or consistently?

Reply:
It's consistently more than 1, and I've got no idea how it wasn't failing before 🤔  
I could alternatively clean approved & pending scripts before each test to make that one work, but again don't see much value added in this assertion.

Reply:
I suspect that scripts saved by non-admins result in two pending scripts since we do not call `ApprovalContext.withKey` here, which is [the only thing](https://github.com/jenkinsci/script-security-plugin/blob/4bd517d6dbf0926c3d506005e11187006d8f556f/src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java#L455-L458) that would prevent multiple pending scripts with the same content since we call `configuring` twice in most cases (constructor + newInstance/readResolve). Maybe we should pass `Run.getExternalizableId()` (maybe with `CpsFlowDefinition.class.getName()` as well?) as the key? `SecureGroovyScript` has [a convenience method](https://github.com/jenkinsci/script-security-plugin/blob/4bd517d6dbf0926c3d506005e11187006d8f556f/src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SecureGroovyScript.java#L164) to use the current `Item` as the key which should prevent this issue for it.

Reply:
Actually, `pendingScripts` is already defined as a set and equality on its members is just the script and the language, so what I noted shouldn't matter at all. Are the pending scripts in this case distinct?

Reply:
I haven't looked at it under debugger. I'll check now.

Reply:
So the other pending one is from another test

Reply:
Ah ok, false alarm then, sorry!

Reply:
Maybe just move this test case to another class not using `@ClassRule`?